js/src/GridHeatMapModel.ts

```diff
@@ -57,22 +58,24 @@ export class GridHeatMapModel extends MarkModel {
     update_data() {
         this.dirty = true;
         // Handling data updates
-        const that = this;
         this.colors = this.get("color");
-        this.rows = this.get("row");
-        this.columns = this.get("column");
 
+        const num_rows = this.colors.length;
         const num_cols = this.colors[0].length;
+
+        this.rows = this.get("row") || _.range(num_rows);
+        this.columns = this.get("column") || _.range(num_cols);
```

Comment:
same here, should we use `num_rows + 1` and `num_cols + 1`?

Reply:
As per in-person discussion. We found that `_.range(num_rows)` is a valid default value because it uses an Ordinal scale by default and rows are centered on ticks (0, 1, 2, 3...), and in this case values in `row` are the values on which rows are centered.